src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/HttpResponseHeadersReadOnlyCollection.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http.Headers;
+
+namespace System.Net.WebSockets
+{
+    internal sealed class HttpResponseHeadersReadOnlyCollection : IReadOnlyDictionary<string, IEnumerable<string>>
+    {
+        private readonly IReadOnlyDictionary<string, HeaderStringValues> _headers;
+
+        public HttpResponseHeadersReadOnlyCollection(HttpResponseHeaders headers) => _headers = headers.NonValidated;
+
+        public IEnumerable<string> this[string key] => _headers[key];
+
+        public IEnumerable<string> Keys => _headers.Keys;
+
+        public IEnumerable<IEnumerable<string>> Values => (IEnumerable<IEnumerable<string>>)_headers.Values;
```

Comment:
I am not sure you can cast like that, because HeaderStringValues is a struct 🤔 I'm also suspicions about GetEnumerator()

Could you please double-check and add tests fot this class, if not for all then at least for Values and GetEnumerator?